src/main_app/su_services/jobs_files_service.py

```
@@ -68,9 +56,34 @@ def load_job_result(result_file: str) -> Dict[str, Any] | None:
         return None
 
 
+def create_job_cancelled_file(filename: str) -> Path:
+    jobs_dir = get_jobs_data_dir()
+    # Use microseconds to avoid race conditions if multiple jobs complete simultaneously
+    filepath = jobs_dir / filename
+    try:
+        with open(filepath, "w", encoding="utf-8") as f:
+            f.write("cancelled")
+        return filepath
+    except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)